R/derive_var_shift.R

```diff
@@ -0,0 +1,109 @@
+#' Derive Shift
+#'
+#' Derives a character shift variable containing concatenated shift in
+#' values based on user-defined pairing, e.g., shift from baseline to
+#' analysis value, shift from baseline grade to analysis grade, ...
+#'
+#' @param dataset Input dataset
+#'
+#'   The columns specified by `from_var` and the `to_var`parameters are expected.
+#'
+#' @param new_var Name of the character shift variable to create.
+#'
+#' @param from_var Variable containing value to shift from.
+#'
+#' @param to_var Variable containing value to shift to.
+#'
+#' @param post_baseline_condition Condition to identify post-baseline records. If not
+#' specified, `new_var` is populated for all records.
+#'
+#'  Default: `NULL`
+#'
+#' @details `new_var` is derived by concatenating the values of `from_var` to values of `to_var`
+#' (e.g. "NORMAL to HIGH"). When `from_var` or `to_var` has missing value, the
+#' missing value is replaced by `NULL` (e.g. "NORMAL to NULL"). If `post_baseline_condition` is
+#' specified, `new_var` is populated only for post-baseline records. If `post_baseline_condition`
+#' is `NULL` (default), `new_var` is populated for all records.
+#'
+#'
+#' @author Annie Yang
+#'
+#' @return The input dataset with the character shift variable added
+#'
+#' @keywords adam bds adlb derivation
+#'
+#' @export
+#'
+#'
+#' @examples
+#' data <- tibble::tribble(
+#' ~USUBJID, ~PARAMCD, ~AVAL, ~ABLFL, ~BNRIND, ~ANRIND,
+#' "P01",    "ALB",     33,    "Y",    "LOW",   "LOW",
+#' "P01",    "ALB",     38,    "",     "LOW",   "NORMAL",
+#' "P01",    "ALB",     NA,    "",     "LOW",   "",
+#' "P02",    "ALB",     37,    "Y",    "NORMAL","NORMAL",
+#' "P02",    "ALB",     49,    "",     "NORMAL","HIGH",
+#' "P02",    "SODIUM",  147,   "Y",    "HIGH",  "HIGH"
+#' )
+#'
+#' library(dplyr, warn.conflicts = FALSE)
```

Comment:
Please put the `library()` call at the top if the example before the data creation.

Reply:
Updated